samples/client/petstore/go/go-petstore/pet_api.go

```diff
@@ -35,89 +35,70 @@ func NewPetApiWithBasePath(basePath string) *PetApi{
  * @param body Pet object that needs to be added to the store
  * @return void
  */
-func (a PetApi) AddPet (body Pet) (error) {
-    // verify the required parameter 'body' is set
-    if &body == nil {
-        return errors.New("Missing required parameter 'body' when calling PetApi->AddPet")
-    }
-    _sling := sling.New().Post(a.Configuration.BasePath)
+func (a PetApi) AddPet (body Pet) (ApiResponse, error) {
```

Comment:
Ah, ok, I thinking I was looking at a different petstore spec that does indeed return a pet object, makes sense now.